GiEnJul/ClientApp/src/components/registerGiver/LocationGiver.tsx

```diff
@@ -1,100 +1,89 @@
+  locationOptions: string[];
 }
-const LocationGiver: React.FC<Props> = ({ nextStep, prevStep, handleLocationChange, values, options, placeHolder }) => {
 
-    const Continue = (e: any) => {
-        e.preventDefault();
-        nextStep();
-    }
-    const classes = useStyles();
+const LocationGiver: React.FC<Props> = ({
+  nextStep,
+  values,
+  handleLocationChange,
+  placeHolder,
+  locationOptions,
+}) => {
+  const classes = useStyles();
 
-    if (prevStep) {
-        const Previous = (e: any) => {
-            e.preventDefault();
-            prevStep();
-        }
-
-        return (
-            <Container key={values.familyType}>
-                <ValidatorForm
-                    onSubmit={Continue}
-                    onError={errors => console.log(errors)}
-                    style={{ width: '100%', marginTop: '20px' }}
-                >
-                    <SelectValidator
-                        variant="outlined"
-                        fullWidth
-                        autoFocus
-                        placeholder={placeHolder}
-                        validators={['required']}
-                        name="familyType-input"
-                        value={values.familyType ? values.familyType : ""}
-                        onChange={handleLocationChange}
-                        label="Familiesammensetning*"
-                        errorMessages={['Hvilken familie venter på din gave?']}
-                    >
-                        {options.map(x =>
-                            <MenuItem key={x} value={x}>{x}</MenuItem>)}
-                    </SelectValidator>
-                    <Grid container spacing={2} justify="center" className={classes.submit}>
-                        <Grid item >
-                            <Button variant="contained" onClick={Previous} >Tilbake</Button>
-                        </Grid>
-                        <Grid item>
-                            <Button variant="contained" type="submit">Neste</Button>
-                        </Grid>
-                    </Grid>
-                </ValidatorForm>
-            </Container>
-        )
-    }
-    return (
-        <Container>
-            <ValidatorForm
-                onSubmit={Continue}
-                onError={errors => console.log(errors)}
-                style={{ width: '100%', marginTop: '20px' }}
-            >
-                <SelectValidator
-                    variant="outlined"
-                    fullWidth
-                    autoFocus
-                    placeholder={placeHolder}
-                    validators={['required']}
-                    label="Lokasjon*"
-                    name="location-input"
-                    value={values.location ? values.location : ""}
-                    id="location-input"
-                    onChange={handleLocationChange}
-                    errorMessages={['Hvor vil du spre glede?']}
+  return (
+    <Container>
+      <ValidatorForm
+        onSubmit={nextStep}
```

Comment:
When a form is to be submitted it should have a onSubmit function. Having a function that can potentially (if theres a bug) lead the user to submit a form without it being complete, can cause problems.

If im not mistaken you guys are "waiting" for the counter to be complete or 5/5 ie. and then you check if everything is valid and then if it is then the information is submitted, correct?